src/main/java/cs2103/v15_1j/jimjim/model/TaskEvent.java

```diff
@@ -0,0 +1,38 @@
+package cs2103.v15_1j.jimjim.model;
+import java.time.LocalDateTime;
+
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+
+public abstract class TaskEvent {
+	private StringProperty name;
+	private IntegerProperty id;
+	
+	public String getName() {
+		return name.get();
+	}
+	
+	public void setName(String name){
+		this.name = new SimpleStringProperty(name);
+	}
+	
+	public StringProperty taskNameProperty() {
+        return name;
+    }
+	
+	public void setID(int id){
+		this.id = new SimpleIntegerProperty(id);
+	}
+	
+	public int getID(){
+		return id.get();
+	}
+	
+	public ObjectProperty<LocalDateTime> dateTimeProperty(){
+		return null;
```

Comment:
Getter and Setter for ID? I need to use the taskNameProperty and SetName Methods at the very least. The ObjectProperty<LocalDateTime> is for PolyMorphism. I forgot the exact case where I had to use it. It would then allow me to use Dynamic Binding (was this the term? the one where it would automatically call the overridden method in the child object)